src/llm_vm/onsite_llm.py

```diff
@@ -303,6 +304,36 @@ def model_loader(self):
         return LlamaForCausalLM.from_pretrained(self.model_uri)
     def tokenizer_loader(self):
         return LlamaTokenizer.from_pretrained(self.model_uri)
+    
+@RegisterModelClass("quantized-llama")
+class Quantized_Llama(Base_Onsite_LLM):
+    """
+    Class for running quantized Llama instances that use GGML
+
+    Attributes:
+        model_uri (str): Hugging Face Endpoint for LLM
+        model (LLM): The large language model. CTransformers includes its own tokenizer in the model
+
+    Methods:
+        model_loader: Loads the LLM into memory
+        tokenizer_loader: Does nothing. CTransformers includes its own tokenizer so this is unnecessary
+        generate: Generates a response from a given prompt with the loaded LLM and tokenizer
+    """
+    model_uri="TheBloke/LLaMa-7B-GGML"
+
+    def model_loader(self):
+        # This file specifically is the smallest model.
+        return AutoModelForCausalLM.from_pretrained(self.model_uri, model_file="llama-7b.ggmlv3.q2_K.bin")
+
+    # CTransformers loads its tokenizer in the model, so this function is unnecessary
+    def tokenizer_loader(self):
+        return
+
+    def generate(self, prompt, max_length=100, **kwargs):
+        inputs = self.model.tokenize(prompt)
+        generated_tokens = self.model.generate(inputs)
```

Comment:
shouldnt we pass in kwargs? i think i have some example where we do that, or is that still a massive TODO fixme?

Reply:
I'll add kwargs, some classes like the GPT ones take kwargs but not sure how fleshed out those implementations are